fix: Locale overrides should only allow one row per machine

liordino opened the pull request:
fix: filter already overriden machines on locales

liordino committed:
fix: filter already overriden machines on locales

liordino commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Locale overrides should only allow one row per machine](https://trello.com/c/5IvfE8nD/190-locale-overrides-should-only-allow-one-row-per-machine)

RafaelTaranto commented on new-lamassu-admin/src/pages/Locales/Locales.js:
```diff
@@ -69,6 +69,18 @@ const Locales = ({ name: SCREEN_KEY }) => {
     return saveConfig({ variables: { config } })
   }
 
+  const removeOverridenMachines = data => {
+    if (data) {
+      const overridenMachines = locale.overrides?.map(o => o.machine)
+      const machinesIndex = data.findIndex(o => o.name === 'machine')
+      data[machinesIndex].inputProps.options = data[
+        machinesIndex
+      ].inputProps.options?.filter(m => !overridenMachines.includes(m.deviceId))
+    }
+
+    return data
+  }
+
```
This wont work. 

- empty overrides table break the page.
- already existing overrides won't show the machine when being edited.

Take a look at how the FiatBalanceOverrides does it. Better to just follow that logic.

liordino committed:
fix: moved the filtering of already overriden machines to the locales
helper

fix: allow already existing overrides to show the machine when
being edited

liordino committed:
fix: removed debug logging

RafaelTaranto commented on new-lamassu-admin/src/pages/Locales/helper.js:
```diff
@@ -39,7 +54,11 @@ const allFields = getData => {
+        options: it =>
+          R.concat(
+            suggestionFilter(machineData),
+            findSuggestion(it, machineData)
```
Probably better to do

```
R.concat(it?.machine ? [it.machine] : [])(suggestionFilter...)
```

After the changes on Autocomplete the code checks for id to match a option, so I don't think we need to fetch it within the machineData.

RafaelTaranto commented on new-lamassu-admin/src/pages/Locales/helper.js:
```diff
@@ -20,12 +23,24 @@ const allFields = getData => {
   }
 
+  const overridenMachines = R.map(override => override.machine, auxElements)
+  const suggestionFilter = R.filter(
+    it => !R.contains(it.deviceId, overridenMachines)
+  )
```
Better to use https://ramdajs.com/docs/#differenceWith

liordino committed:
refactor: ramda function replacements to cleanup the code
